src/Grid/topology.jl

```diff
@@ -70,9 +70,14 @@ mutable struct ExclusiveTopology <: AbstractTopology
     edge_edge_neighbor::Matrix{EntityNeighborhood{EdgeIndex}}
     # lazy constructed face topology
     face_skeleton::Union{Vector{FaceIndex}, Nothing}
+    edge_skeleton::Union{Vector{EdgeIndex}, Nothing}
```

Comment:
This seems only valid for 2D and should rather be `facet_skeleton` ?

Reply:
Hmm, I dont think so. My idea is that you generate the edge_skeleton, face_skeleton etc the same for both a 2d and 3d grid (because topologicaly they are definied in the same way now), and then if you want to loop over the dim-1 entity, you would dispatch on the dim-parameter:

```
function facetskeleton(top::ExclusiveTopology, grid::Grid{2})
    return top.edge_skeleton
end
function facetskeleton(top::ExclusiveTopology, grid::Grid{3})
    return top.face_skeleton
end
```

Reply:
I think I'm wondering in general about the dispatch. If I'm not mistaken, then you define a `facet` as a dim-1 entity and you can store directly the `facetskeleton` without having the extra field in the struct

```
function facetskeleton(top::ExclusiveTopology, grid)
    return top.facetskeleton
end
```

Or is there some advantage of having both fields?

Reply:
But what if you have a hex mesh? Then you will have both edge_skeleton and face_skeleton, and you will have a field called facet_skeleton which is identical to face_skeleton? 

I realise now though that the dispatch approach might not work for shells in 3d... the dispatch for facetskeleton(..., grid{3}) would return the face_skeleton, even though you probably would want egde_skeleton...

Reply:
Ah okay! I think as of now, there is no construction of edge skeleton in 3D cases planned. Not sure if they are useful in some cases, but if we want to include them, then this makes sense